### PR TITLE
Only generate methods with supported types

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
We currently generate code that isn't supported by the detox client.
This causes confusion amongst developers adopting the generated code (see #228).

### CODE REVIEW

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 1]
```diff
@@ -79,54 +79,6 @@ simulate a long press.
     };
   }
 
-  /*Returns an action that holds down finger for specified @c duration to simulate a long press.
-
-@param duration The duration of the long press.
-
-@return A GREYAction that performs a long press on an element.
-*/static actionForLongPressWithDuration(duration) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: duration is `CFTimeInterval` === `double`, no need to kill it

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 2]
```diff
@@ -196,60 +148,6 @@ starting from the given start points.
     };
   }
 
-  /*@return A GREYAction that scrolls to the given content @c edge of a scroll view.
-*/static actionForScrollToContentEdge(edge) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: edge is `GREYContentEdge` , an ENUM, returns an NSString

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 3]
```diff
@@ -196,60 +148,6 @@ starting from the given start points.
...
-
-@param edge                   The edge towards which the scrolling is to take place.
-@param xOriginStartPercentage X coordinate of the start point specified as a percentage (0, 1)
-exclusive, of the total width of the scrollable visible area.
-@param yOriginStartPercentage Y coordinate of the start point specified as a percentage (0, 1)
-exclusive, of the total height of the scrollable visible area.
-
-@return A GREYAction that scrolls to the given content @c edge of a scroll view with the scroll
-action starting from the given start point.
-*/static actionForScrollToContentEdgeXOriginStartPercentageYOriginStartPercentage(edge, xOriginStartPercentage, yOriginStartPercentage) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: edge is an NSInteger

Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: Probably needs to be normalized like you did before

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 4]
```diff
@@ -484,105 +382,6 @@ direction from the specified point.
...
 
-  /*Returns an action that pinches view quickly in the specified @c direction and @c angle.
-
-@param  pinchDirection The direction of the pinch action.
-@param  angle          The angle of the pinch action in radians.
-Use @c kGREYPinchAngleDefault for the default angle (currently set to
-30 degrees).
-
-@return A GREYAction that performs a fast pinch on the view in the specified @c direction.
-*/static actionForPinchFastInDirectionWithAngle(pinchDirection, angle) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: direction is NSInteger

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 5]
```diff
@@ -676,116 +475,6 @@ For Example: @"Helpo\b\bloWorld" will type HelloWorld in Objective-C.
     };
   }
 
-  /*Returns an action that toggles a switch control. This action is applicable to all elements that
-implement the selector UISwitch::isOn and include UISwitch controls.
-
-@param on The switch control state.
-
-@return A GREYAction to toggle a UISwitch.
-*/static actionForTurnSwitchOn(on) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: yea, we have problems with bools

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 6]
```diff
@@ -676,116 +475,6 @@ For Example: @"Helpo\b\bloWorld" will type HelloWorld in Objective-C.
...
-      }]
-    };
-  }
-
-  /*Returns an action that selects @c value on the given @c column of a UIPickerView.
-
-@param column The UIPickerView column being set.
-@param value  The value to set the UIPickerView.
-
-@return A GREYAction to set the value of a specified column of a UIPickerView.
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: why was this removed?

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 7]
```diff
@@ -676,116 +475,6 @@ For Example: @"Helpo\b\bloWorld" will type HelloWorld in Objective-C.
...
-      }]
-    };
-  }
-
-  /*Returns an action that injects dates/times into UIDatePickers.
-
-@param date The date to set the UIDatePicker.
-
-@return A GREYAction that sets a given date/time on a UIDatePicker.
-*/static actionForSetDate(date) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: :(

File: detox/src/ios/earlgreyapi/GREYActions.js [hunk 8]
```diff
@@ -79,54 +79,6 @@ simulate a long press.
...
-  }
-
-  /*Returns an action that holds down finger for specified @c duration at the specified @c point
-(interpreted as being relative to the element) to simulate a long press.
-
-@param point    The point that should be tapped.
-@param duration The duration of the long press.
-
-@return A GREYAction that performs a long press on an element.
-*/static actionForLongPressAtPointDuration(point, duration) {
```
Comment (rotemmiz) [inline detox/src/ios/earlgreyapi/GREYActions.js]: same goes here

File: detox/src/ios/earlgreyapi/GREYActions.js (hunk 1 above)
Comment (DanielMSchmidt) [inline detox/src/ios/earlgreyapi/GREYActions.js]: Great, I will add this to the list

File: detox/src/ios/earlgreyapi/GREYActions.js (hunk 2 above)
Comment (DanielMSchmidt) [inline detox/src/ios/earlgreyapi/GREYActions.js]: I would like to sanitize these enums. As the sanitizer isn't build yet I excluded them. My next goal is to readd these functions

File: detox/src/ios/earlgreyapi/GREYActions.js (hunk 6 above)
Comment (DanielMSchmidt) [inline detox/src/ios/earlgreyapi/GREYActions.js]: This was 100% a mistake!

File: detox/src/ios/earlgreyapi/GREYActions.js (hunk 1 above)
Comment (DanielMSchmidt) [inline detox/src/ios/earlgreyapi/GREYActions.js]: @rotemmiz I double checked it, it is not included in the [switch statement in method invocation](https://github.com/wix/detox/blob/master/detox/ios/Detox/MethodInvocation.m#L34) Therefore we need to remove it for now. I will add this functionality while extending `MethodInvocation` in the same manner. That way the commits are sane and make sense 👍

**Comment by DanielMSchmidt** [comment]
Added the tests and verified the comments you made. These functions should mostly be unsupported and I would add them together with the change in the iOS mapping in MethodInvocation as we add new functionality. <3 I would merge this as soon as I see the tests green, we can still make changes afterwards ;)
